src/index.js

```diff
@@ -1,49 +1,118 @@
-function parseArticle($, article) {
-  const $article = $(article)
-  const title = $article.find('h3 a').attr('title')
-  return {
-    title,
-    price: normalizePrice($article.find('.price_color').text()),
-    url: `${baseUrl}/${$article.find('h3 a').attr('href')}`,
-    // when it finds a fileurl attribute, saveFiles will save this file to the cozy with a filename
-    // name
-    fileurl: `${baseUrl}/${$article.find('img').attr('src')}`,
-    filename: `${title}.jpg`
+async function login (fields) {
+  log('info', 'Logging in')
+  await request(`${baseUrl}/LoginMDP`)
+  let $
+  try {
+    $ = await request({
+      method: 'POST',
+      uri: `${baseUrl}/LoginMDP?op=c&url=`,
+      form: {
+        url: '',
+        LMDP_Spi: fields.login,
+        LMDP_Password: fields.password,
+        LMDP_Spi_tmp: fields.login,
+        LMDP_Password_tmp: fields.password
+      }
+    })
+  } catch (err) {
+    log('error', 'Website failed while trying to login')
+    log('error', err.message)
+    throw new Error(errors.VENDOR_DOWN)
+  }
+
+  const erreurs = $('.erreur:not(.pasvisible)')
+  if (erreurs.length) {
+    log('error', erreurs.eq(0).text().trim())
+    throw new Error(errors.LOGIN_FAILED)
   }
 }
 
-// convert a price string to a float
-function normalizePrice(price) {
-  return parseFloat(price.trim().replace('£', ''))
+async function fetch () {
+  log('info', 'Fetching the list of documents')
+  let $ = await request(`${baseUrl}/acces-usager/cfs`)
+
+  // get the "Mes documents" link
+  const documentsLink = $('img[name=doc]').closest('a').attr('href')
+  const urlPrefix = documentsLink.split('/')[1] // gets "cesu-XX" from the url
+  $ = await request(`${baseUrl}${documentsLink}`)
+
+  const $form = $('form[name=documentsForm]')
+  const formUrl = $form.attr('action')
+  const token = $form.find('input[name=CSRFTOKEN]').val()
+
+  $ = await request({
+    method: 'POST',
+    uri: `${baseUrl}${formUrl}`,
+    form: {
+      annee: 'all',
+      CSRFTOKEN: token,
+      method: 'rechercheDocuments',
+      typeDocument: 'toutDocument',
+      typeImpot: 'toutImpot'
+    }
+  })
+
+  const documents = scrape($, {
+    fileurl: {
+      attr: 'onclick',
+      parse: onclick => {
+        const viewerUrl = onclick.match(/\((.*)\)/)[1].split(',')[0].slice(1, -1)
+        return `${baseUrl}/${urlPrefix}/${viewerUrl}`
+      }
+    },
+    name: {
+      fn: link => {
+        return $(link).closest('tr').text()
+      }
+    }
+  }, '.cssLienTable')
+
+  log('info', `Found ${documents.length} documents to download`)
+
+  const result = []
+  for (let doc of documents) {
+    log('debug', `Fetching doc url for ${doc.name}`)
+    const $ = await request(doc.fileurl)
+    result.push({
+      fileurl: `${baseUrl}${$('iframe').attr('src')}`,
+      name: doc.name
+    })
+  }
+
+  return result.map(doc => {
+    if (doc.fileurl.match(/ConsultAR/)) {
```

Comment:
Maybe we could check this in the previous `for` block, so we don't need to loop twice? Or did I miss something?